File: solutions/c/grains/1/grains.c
#include "grains.h"

#define SQUARES_COUNT 64

uint64_t square(uint8_t index) {
    if (index < 1 || index > SQUARES_COUNT) {
        return 0;
    }
    return (uint64_t)1 << (index - 1);
}

uint64_t total(void) {
    return UINT64_MAX;
}

File: solutions/elixir/grains/1/lib/grains.ex
defmodule Grains do
  use Bitwise

  def square(n) when n < 1 or n > 64,
    do: {:error, "The requested square must be between 1 and 64 (inclusive)"}
  def square(n), do: {:ok, 1 <<< (n - 1)}
  def total,     do: {:ok, (1 <<< 64) - 1}
  
end
